### PR TITLE
datamash: improve grouping column regex and add csv support

### PR DESCRIPTION
**Pull request by bernt-matthias**
fixes https://github.com/galaxyproject/tools-iuc/issues/4265

downside is that ranges are not possible (e.g. `1-4`) .. but they also were not so far because of the regex validator

plus:

- reordered attributes
- renamed tokens, e.g. TOOL_VERSION
- doc why sorting may be relevant (fixes https://github.com/galaxyproject/tools-iuc/issues/1205)

FOR CONTRIBUTOR:
* [x] - I have read the [CONTRIBUTING.md](https://github.com/galaxyproject/tools-iuc/blob/master/CONTRIBUTING.md) document and this tool is appropriate for the tools-iuc repo.
* [x] - License permits unrestricted use (educational + commercial)
* [ ] - This PR adds a new tool or tool collection
* [x] - This PR updates an existing tool or tool collection
* [ ] - This PR does something else (explain below)

### CODE REVIEW

**Comment by wm75** [comment]
Another downside of this is that you won't be able to control column ordering in the output anymore, would you?

Grouping by 1,2 is not the same as grouping by 2,1.

**Comment by wm75** [comment]
I commented in #1205 why I think making datamash's internal sorting the default could create new surprises.

The only thing I'd really change here is emphasize the impact of sorting more, and I guess to this end I'd:
1. move up the sort input switch to right underneath the grouping fields
2. explain the consequences of sorting/not sorting by example in the tool help section, and point to it from the sort switch help

**Comment by bernt-matthias** [comment]
Good points @wm75. will implement them as soon as possible.

**Comment by bernt-matthias** [comment]
Should be done. 

In addition I added support for all tabular types without running an implicit converter. The difference to the state before is that now the output is csv if the input is csv. Before the output was always tabular since the input was implicitly converted tabular.

We could change this by using `--output-delimiter=x`, i.e. always output tabular.

**Comment by wm75** [comment]
Nice! I like the new csv handling!

Still, I don't think the new way to select the columns for grouping allows for output column sorting, which is rather often what I'd want. If after running datamash I have to run `Cut` and enter the column indices there, I haven't gained anything.

**Comment by bernt-matthias** [comment]
> Still, I don't think the new way to select the columns for grouping allows for output column sorting, which is rather often what I'd want.

Do you have an example from which I could construct a test. I do not really get your point.

**Comment by wm75** [comment]
Hmm, not the most simple example, but taken straight from our SARS-CoV-2 bot runs:
https://usegalaxy.eu/u/sars-cov2-bot/h/gx-surveillance-estcov155---reporting

Look at collections 143 and 178 and how datamash is used both for collapsing lines in a tabular variant report and for moving the last column of the input into its intended place. In general, you often want to first calculate some new column(s) for datamash to work/group on, but e.g. "Compute an expression on every row" will append the new column(s) at the end.

The issue is a bit like the one for the two Cut tools: one requires you to type a list of all column indices you want, but lets you rearrange at the same time, the other one lets you conveniently click together the columns you want, but with no control over their order.

**Review by bgruening — APPROVED** [review]
I think this is ok now. I think the entire problem here is that the data_column UI does not allow for random sorting. This is a UI problem I think and if we can enhance this widget to support this feature here ... btw. same goes for repeat elements they should be sortable

**Review by wm75 — APPROVED** [review]
Thanks @bernt-matthias! Small changes, but together they make the tool quite a bit more user-friendly :)

Please fix the indentation as indicated, but ready to merge otherwise.

File: tools/datamash/datamash-ops.xml [hunk 1]
```diff
@@ -25,15 +25,23 @@
...
+            <sanitizer invalid_char="">
+                <valid initial="string.digits">
+                    <add value="," />
+                </valid>
+                <mapping initial="none">
+                    <add source=" " target=""/>
+                </mapping>
+            </sanitizer>
+            <validator message="Invalid value in field. Allowed is a comma separated list of integer values or the empty string" type="regex">(^$)|(^\s*\d+\s*(,\s*\d+\s*)*$)</validator>
+       </param>
```
[the remark this answers is not in the document]
Comment (bernt-matthias) [reply tools/datamash/datamash-ops.xml]: Ups. Done

**Pull request merged**